dbt_common/contracts/config/materialization.py

```diff
@@ -0,0 +1,11 @@
+from dbt_common.dataclass_schema import StrEnum
+
+
+class OnConfigurationChangeOption(StrEnum):
```

Comment:
I ran into this yesterday. I think this might belong in `dbt-adapter`, but open to thoughts.

Reply:
I thought this through more. `dbt-adapter` needs to know how to make decisions based on the value of this config, but this is config that's available regardless of the adapter. I'm fine with it living here.